### PR TITLE
DBZ-1658 Allowing connector tests to run with downstream dependencies

### PR DESCRIPTION
**Pull request by jcechace**

### CODE REVIEW

**Comment by jpechane** [comment]
@jcechace Please remove jobs managed by job seeder from jenkins job builder so we will not step on each other toe

**Comment by jcechace** [comment]
@jpechane just to clarify. Do you mean removing the YAML JJB configs for jobs which do have a Job DSL version now?

**Comment by jpechane** [comment]
@jcechace Yes

File: jenkins-jobs/job-dsl/mongodb_matrix_test.groovy [hunk 1]
```diff
@@ -44,8 +43,25 @@ matrixJob('debezium-mongodb-matrix-test') {
...
+    PROFILE_PROD="none"
+    git clone $REPOSITORY . 
+    git checkout $BRANCH
+fi
+                    
+# Run maven build
+mvn clean install -U -s $HOME/.m2/settings-snapshots.xml -pl debezium-connector-mongodb -am -fae \
+    -Dmaven.test.failure.ignore=true \
+    -Dversion.mongo.server=$MONGODB_VERSION \
+    -P$PROFILE_PROD
```
Comment (jpechane) [inline jenkins-jobs/job-dsl/mongodb_matrix_test.groovy]: @jcechace  What are `pnc` and `none` profiles?

Comment (jcechace) [reply jenkins-jobs/job-dsl/mongodb_matrix_test.groovy]: pnc is a profile set in settings.xml (available in our jenkins) containing indy repository and plugin repository. None is a non-exsistent profile (ignored by maven) used to set the variable if the build is not product-based.

**Pull request merged**